adapters/platform_win32.c

```diff
@@ -96,21 +155,43 @@ STRING_HANDLE platform_get_platform_info(void)
         DWORD product_type;
         if (GetProductInfo(osvi.dwMajorVersion, osvi.dwMinorVersion, 0, 0, &product_type))
         {
-            result = STRING_construct_sprintf("(native; WindowsProduct:0x%08x %d.%d; %s)", product_type, osvi.dwMajorVersion, osvi.dwMinorVersion, arch);
+            result = STRING_construct_sprintf("(native; WindowsProduct:0x%08x %d.%d; %s", product_type, osvi.dwMajorVersion, osvi.dwMinorVersion, arch);
         }
     }
 
     if (result == NULL)
     {
         DWORD dwVersion = GetVersion();
-        result = STRING_construct_sprintf("(native; WindowsProduct:Windows NT %d.%d; %s)", LOBYTE(LOWORD(dwVersion)), HIBYTE(LOWORD(dwVersion)), arch);
+        result = STRING_construct_sprintf("(native; WindowsProduct:Windows NT %d.%d; %s", LOBYTE(LOWORD(dwVersion)), HIBYTE(LOWORD(dwVersion)), arch);
     }
 #pragma warning(default:4996)
 
     if (result == NULL)
     {
         LogError("STRING_construct_sprintf failed");
     }
+
+    if (options & PLATFORM_OPTION_RETRIEVE_SQM)
+    {
+        // Failure here should continue
+        char* sqm_info = get_win_sqm_info();
+        if (sqm_info != NULL)
+        {
+            if (STRING_sprintf(result, "; %s)", sqm_info) != 0)
```

Comment:
Usually when there is a failure in the SDK we have been destroying whatever data that has been put together before in the function, and returning NULL.
Wondering which way would be preferred here. Looks like the way implemented here is better, but what Yi has to comment on this? Is partial info really better than no info for you?

Reply:
Yes, SQM ID is optional - if it exists we'd like to have it so we can join it with other telemetry we collect from Windows.  If it doesn't exist, it should at least return the default values in the string.  It's not a hard requirement that SQM ID needs to be there.

Reply:
But it looks like there's an issue here since the parentheses aren't closed if the SQM ID isn't found.  I'll fix that.